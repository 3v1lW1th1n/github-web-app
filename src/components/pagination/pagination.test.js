import 'jest-styled-components'
import { act } from 'react-dom/test-utils'

import React from 'react'
import { shallow, mount } from 'enzyme'
import Pagination from './'
import repos from '../../mock/repositories'

describe('<Pagination />', () => {
  it('should match default snapshot', () => {
    const component = shallow(<Pagination />)
    expect(component).toMatchSnapshot()
  })

  it('should have the right displayName', () => {
    expect(Pagination.displayName).toEqual('Pagination')
  })
  it('should paginate the items correctly', () => {
    const props = {
      initialPage: 1,
      pageSize: 3
    }
    const mockFn = jest.fn()
    const component = mount(<Pagination
      items={repos}
      initialPage={props.initialPage}
      pageSize={props.pageSize}
      onChangePage={mockFn} />
    )
    expect(mockFn.mock.calls[0][0].length).toBe(props.pageSize)
    expect(mockFn.mock.calls[0][1]).toBe(1)
    act(() => {
      component.find('#next-page').prop('onClick')()
    })
    expect(mockFn.mock.calls[1][0].length).toBe(repos.length - props.pageSize)
    expect(mockFn.mock.calls[1][1]).toBe(2)
  })
  it('should redirect to page 1 if current page is grater than total pages', () => {
    const mockFn = jest.fn()
    const component = mount(<Pagination
      items={repos}
      initialPage={10}
      onChangePage={mockFn} />
    )
    component.update()
    expect(mockFn.mock.calls[0][1]).toBe(1)
  })
})
