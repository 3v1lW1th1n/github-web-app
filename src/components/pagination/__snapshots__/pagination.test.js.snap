// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match default snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "initialPage": 1,
        "items": Array [],
        "pageSize": 6,
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "StyledPagination-sc-7bug5c-0",
        "isStatic": true,
        "rules": Array [
          "width:100%;text-align:center;font-family:var(--site-font);.pagination{padding:16px 0;display:inline-flex;}.page-item{width:32px;height:32px;border-radius:4px;display:flex;align-items:center;justify-content:center;cursor:pointer;border:1px solid transparent;:hover{border:1px solid rgba(0,0,0,.5);}}.page-item + .page-item{margin-left:8px;}.active{border:1px solid rgba(0,0,0,.5);font-size:24px;}.disabled{color:rgba(0,0,0,.3);pointer-events:none;}",
        ],
      },
      "displayName": "Pagination",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "StyledPagination-sc-7bug5c-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  initialPage={1}
  items={Array []}
  pageSize={6}
/>
`;
